Jakarta - EE 9 - Serialization issue in MongoDB IT PojoResourceTest

The test is failing with:

```
2022-08-25 15:09:49,202 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /pojos failed, error id: 76c5f6d0-f5e4-475f-8cd7-dfa13a436fc7-1: jakarta.json.bind.JsonbException: Unable to deserialize property 'optionalString' because of: Incorrect position for processing type: class java.lang.String. Received event: START_OBJECT Allowed: [VALUE_NUMBER, VALUE_FALSE, VALUE_TRUE, VALUE_NULL, VALUE_STRING]
	at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:80)
	at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:31)
	at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:57)
	at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:29)
	at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
	at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
	at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
	at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
	at org.eclipse.yasson.internal.DeserializationContextImpl.deserializeItem(DeserializationContextImpl.java:138)
	at org.eclipse.yasson.internal.DeserializationContextImpl.deserialize(DeserializationContextImpl.java:127)
	at org.eclipse.yasson.internal.JsonBinding.deserialize(JsonBinding.java:55)
	at org.eclipse.yasson.internal.JsonBinding.fromJson(JsonBinding.java:95)
	at org.jboss.resteasy.reactive.server.jsonb.JsonbMessageBodyReader.doReadFrom(JsonbMessageBodyReader.java:55)
	at org.jboss.resteasy.reactive.server.jsonb.JsonbMessageBodyReader.readFrom(JsonbMessageBodyReader.java:37)
	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.readFrom(RequestDeserializeHandler.java:108)
	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:67)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:102)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:140)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:557)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

If I remove the `Optiona<String>` in `Pojo`, it works OK but as this field is supposed to test something specific, we can't really do that.

Not sure if it's due to a Yasson regression or a bug in what we're doing, it needs some digging. We are using the latest version of Yasson.

This issue is strange, when I launch the project locally and test it with curl I saw a NPE inside Yasson (which should never occurs).

```
Caused by: java.lang.NullPointerException
	at org.eclipse.yasson.internal.serializer.OptionalSerializer.serialize(OptionalSerializer.java:36)
	at org.eclipse.yasson.internal.serializer.ValueGetterSerializer.serialize(ValueGetterSerializer.java:43)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.lambda$serialize$0(ObjectSerializer.java:41)
```

The OptionalSerializer code looks like this, the NPE arrises at the `delegate.serialize(optional.orElse(null), generator, context);` according to me it can only occurs if optional is null which of course is valid (even if bad practice). I don't know if it's normal at this stage (there is a nullserializer that may catch this) but it definitly looks like a bug at yasson side.

```
    @SuppressWarnings("unchecked")
    @Override
    public void serialize(Object value, JsonGenerator generator, SerializationContextImpl context) {
        Optional<Object> optional = (Optional<Object>) value;
        delegate.serialize(optional.orElse(null), generator, context);
```

--- a/integration-tests/mongodb-client/src/test/java/io/quarkus/it/mongodb/PojoResourceTest.java
+++ b/integration-tests/mongodb-client/src/test/java/io/quarkus/it/mongodb/PojoResourceTest.java
@@ -7,9 +7,8 @@ import java.util.Optional;
 
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.condition.DisabledOnOs;
-import org.junit.jupiter.api.condition.OS;
 
 import io.quarkus.it.mongodb.pojo.Pojo;
 import io.quarkus.test.common.QuarkusTestResource;
@@ -20,7 +19,12 @@ import io.restassured.response.Response;
 
 @QuarkusTest
 @QuarkusTestResource(MongoTestResource.class)
-@DisabledOnOs(OS.WINDOWS)
+//@DisabledOnOs(OS.WINDOWS)
+// This test is disabled but should only be disable on Windows (as with other MongoDB tests) due an issue at Yasson
+// side https://github.com/eclipse-ee4j/yasson/issues/575.
+// The Jakarata EE migration upgrades to Yasson 3.0.1 that contains the issue so we temporarily disable this test.
+// To avoid forgetting it forever an issue has been created, see https://github.com/quarkusio/quarkus/issues/27619.
+@Disabled
 public class PojoResourceTest {
 
     @BeforeEach

